sdcm/gce_log_monitor.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import time
+from typing import Dict
+
+from dateutil import parser
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+
+from sdcm.keystore import KeyStore
+from sdcm.sct_events import Severity
+from sdcm.sct_events.base import SctEvent, EventPeriod
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class GceLoggingClient:  # pylint: disable=too-few-public-methods
+
+    def __init__(self):
+        raw_credentials = KeyStore().get_logging_gcp_credentials()
+        self.credentials = service_account.Credentials.from_service_account_info(raw_credentials)
+        self.project_id = raw_credentials['project_id']
+
+    def get_error_logs(self, since: float):
+        """Gets 'hostError' and 'automaticRestart' logs entries from GCE since the time provided in argument.
+        Returns list of entries in a form of dictionaries. See example output in unit tests."""
+        since = time.strftime('%Y-%m-%dT%H:%M:%SZ', time.gmtime(since))
+        with build('logging', 'v2', credentials=self.credentials, cache_discovery=False) as service:
+            return self._get_log_entries(service, since)
+
+    def _get_log_entries(self, service, since, page_token=None):
+        body = {
+            "resourceNames": [
+                f"projects/{self.project_id}"
+            ],
+            "filter": f'protoPayload.methodName="compute.instances.automaticRestart" '
```

Comment:
this filter isn't good enough, since it's doesn't cross match with the instances we are using.

so it can detect other test suffering from this issue.